data/org.freedesktop.portal.Notification.xml

```diff
@@ -66,21 +68,69 @@
 
+        * ``markup-body`` (``s``)
+
+          The same as ``body`` but with support for markup formatting.
+          The markup is XML-based and supports a small subset of HTML
+          ``<b>...</b>``, ``<i>...</i>`` and ``<a href="...">...</a>``.
+
+          Any markup not supported, e.g. new lines, will be removed from
+          the string. In the future, the set of supported markup may be extended.
+
+          This can be a long string but note that it may not be fully displayed.
+
         * ``icon`` (``v``)
 
-          Serialized icon (see `g_icon_serialize()
-          <https://docs.gtk.org/gio/method.Icon.serialize.html>`_).
+          A serialized icon to add to the notification. The format for
+          serialized icon is a tuple (sv) with the following supported keys:
+
+          * ``themed`` (``as``)
+
+            A themed icon containing an array of strings with the icon names.
+
+            This is the same format as a serialized `GThemedIcon
+            <https://docs.gtk.org/gio/class.ThemedIcon.html>`_.
+
+          * ``bytes`` (``ay``)
 
-          The portal only accepts serialized GThemedIcon and GBytesIcons. Both of
-          these have the form (sv). For themed icons, the string is "themed",
-          and the value is an array of strings containing the icon names.
-          For bytes icons, the string is "bytes", and the value is a bytestring
-          containing the icon data in png, jpeg or svg form. For historical
-          reasons, it is also possible to send a simple string for themed
-          icons with a single icon name.
+            A bytes icon containing a bytestring with the icon data in
+            png, jpeg or svg form.
+
+            This is the same format as a serialized `GBytesIcon
+            <https://docs.gtk.org/gio/class.BytesIcon.html>`_.
+
+          * ``file-descriptor`` (``h``)
+
+            A file descriptor to an image file in png, jpeg or svg form.
+
+          For historical reasons, it is also possible to send a simple string
+          for themed icons with a single icon name.
 
           There may be further restrictions on the supported kinds of icons.
 
+        * ``sound`` (``v``)
+
+          A serialized sound to add to the notification.
+          Supported sound formats are oga and wav.
+
+          The format for
+          serialized sound is a tuple (sv) with the following supported keys:
+
+          * ``bytes`` (``ay``)
+
+            A bytes sound containing a bytestring with the sound data.
+
+          * ``file-descriptor`` (``h``)
+
+            A file descriptor to a sound file.
+
+          To play the default sound the string ``default`` can be passed.
+          To play no sound at all the string ``silent`` can be passed.
+          If this property isn't specified the notification server can decide
+          whether to play a sound.
+
+          There may be further restrictions on the supported kinds of sounds.
```

Comment:
Supported formats are wav and oga. I mirrored the wording from the icon property. In case of icons there are a couple of restrictions e.g. size but that's always the same since it't implemented by xdg-desktop-portal and not by notification servers.

Reply:
Would be really nice for that to be written, I imagine e.g. KDE to not to support sounds at all if not required as it currently doesn't support the fdo's sound capability and all KDE applications play sounds client-side via libcanberra.

Reply:
Playing sound will be required to be implemented by the notification server or portal backend.  We really want that application developers can assume a consistent feature set across different desktops.

Reply:
I hope you're right... But I feel this will end up with not all backends implementing this and the feature become unreliable